### ISSUE TITLE
[Bug] Parallel assignment going wrong 2

### ISSUE THREAD

**Issue by squarefk** [taichi-dev/taichi#1471]
```
import taichi as ti


ti.init(arch=ti.cpu, print_preprocessed=True)
real = ti.f32
mat = ti.var(real, shape=(3, 16))
cnt = ti.var(dt=ti.i32, shape=())


@ti.kernel
def do_something():
    cnt[None] = 0
    for r in range(4):
        for i in ti.static(range(4)):
            c = ti.atomic_add(cnt[None], 1)
            mat[0, c], mat[1, c], mat[2, c] = 1.0, 2.0, 3.0
    for c in range(16):
        print(mat[0, c], mat[1, c], mat[2, c])


if __name__ == "__main__":
    do_something()
```

**Comment by archibate** [comment]
Excuse me? What's wrong? On my end:
```
0.000000 0.000000 0.000000
0.000000 0.000000 0.000000
0.000000 0.000000 0.000000
1.000000 2.000000 3.000000
0.000000 0.000000 0.000000
0.000000 0.000000 0.000000
0.000000 0.000000 0.000000
1.000000 2.000000 3.000000
0.000000 0.000000 0.000000
0.000000 0.000000 0.000000
0.000000 0.000000 0.000000
1.000000 2.000000 3.000000
0.000000 0.000000 0.000000
0.000000 0.000000 0.000000
0.000000 0.000000 0.000000
1.000000 2.000000 3.000000
```
Is this our expected output?

**Comment by xumingkuan** [comment]
I think `(1.000000 2.000000 3.000000)*16` is expected, and it's fixed in #1473.
